quantmsrescore/alphapeptdeep.py

```
@@ -0,0 +1,931 @@
+    for spectrum in read_spectrum_file(spec_file):
+
+        # Match spectrum ID with provided regex
+        match = spectrum_id_regex.search(spectrum.identifier)
+        try:
+            spectrum_id = match[1]
+        except (TypeError, IndexError):
+            raise exceptions.TitlePatternError(
+                f"Spectrum title pattern `{spectrum_id_pattern}` could not be matched to "
+                f"spectrum ID `{spectrum.identifier}`. "
+                " Are you sure that the regex contains a capturing group?"
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)